### PR TITLE
Fix modal dialog max-width and text wrapping

### PR DESCRIPTION
**Pull request by tetchel**
Signed-off-by: Tim Etchells <timetchells@ibm.com>

#### What it does

Fixes https://github.com/eclipse-theia/theia/issues/8071
- use `p` instead of `pre` so that the text will wrap to fit in the modal, but add `white-space: pre-wrap` so that newlines are still preserved and spaces are not collapsed
- adds max-width and max-height to modal

#### How to test

Install the vsix from https://github.com/tetchel/theia-modal-wrap and run the command it contributes

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Screenshot:
image

### CODE REVIEW

**Comment by tetchel** [comment]
@vince-fugnitto

**Comment by kittaakos** [comment]
I do not know if this is out of context or not, but the button color does not look good. Please compare the `New File, `Delete`, and the modal notification dialog.

Screen Shot 2020-06-25 at 15 16 50
Screen Shot 2020-06-25 at 15 17 01
Screen Shot 2020-06-25 at 15 16 43

**Comment by kittaakos** [comment]
The _Close_ button should be `Close` and not `close`. I am going to open a follow-up...

**Comment by tetchel** [comment]
@kittaakos I agree but didn't want to feature-creep my PR. In VS Code, it's `Cancel` instead of `close`. I think `Cancel` is better and prefer it for consistency so I don't need [this](https://github.com/eclipse/codewind-vscode/blob/299cbdca875887e3d4367bb9d30808810415ac13/dev/src/command/project/RemoveProjectCmd.ts#L60)

**Comment by tetchel** [comment]
Let me see about the button colour.

**Comment by kittaakos** [comment]
> I agree but didn't want to feature-creep my first PR.

You do not have to, what you've done is pretty slick so far. :+1: I found another issue, so I will create a follow-up anyway:  the _Command Palette_ should be disabled when there is a model dialog.

But regarding this PR, I would propose one single change, you set a `max-height`, and when the model content is really really long, then it looks ugly:

Screen Shot 2020-06-25 at 15 40 21

I propose the following change:
```diff
diff --git a/packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css b/packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css
index 4cbfa2390..619dcfda2 100644
--- a/packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css
+++ b/packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css
@@ -29,7 +29,6 @@
     min-width: 200px;
     max-width: min(66vw, 800px);
     min-height: 35px;
-    max-height: min(66vh, 600px);
     background-color: var(--theia-editorWidget-background);
     margin-bottom: 1px;
     color: var(--theia-editorWidget-foreground);
@@ -67,7 +66,7 @@
     user-select: text;
     align-self: center;
     flex: 1 100%;
-    height: 100%;
+    max-height: min(66vh, 600px);
     padding: 10px;
     overflow: auto;
     white-space: pre-wrap;
```

And then you can scroll the content:
screencast 2020-06-25 15-39-31

I am not a CSS wizard; what do you think?

**Comment by tetchel** [comment]
for sure, i missed that `height: 100%`

File: packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css [hunk 1]
```diff
@@ -26,10 +26,10 @@
     clear: both;
     box-sizing: border-box;
     position: relative;
-    width: 100%;
     min-width: 200px;
-    background-color: var(--theia-editorWidget-background);
+    max-width: min(66vw, 800px);
     min-height: 35px;
+    background-color: var(--theia-editorWidget-background);
```
Comment (kittaakos) [inline packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css:32]: This is an unnecessary change. Please restore to the [original](https://github.com/eclipse-theia/theia/pull/8080/files#diff-3ff0cf7e13a83017520974a95117399eL31).

File: packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css [hunk 2]
```diff
@@ -60,22 +60,22 @@
...
-    max-height: calc(100vh - 100px);
-    max-width: calc(100vw - 100px);
     -webkit-user-select: text;
     -moz-user-select: text;
     -ms-user-select: text;
     user-select: text;
     align-self: center;
     flex: 1 100%;
-    height: 100%;
+    max-height: min(66vh, 600px);
```
Comment (kittaakos) [inline packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css:69]: Instead of modifying `height` to `max-height`, can you remove it and editor `max-height` [above](https://github.com/eclipse-theia/theia/pull/8080/files#diff-3ff0cf7e13a83017520974a95117399eL63).

Comment (tetchel) [reply packages/plugin-ext/src/main/browser/dialogs/style/modal-notification.css:69]: For both of these I did it this way because I prefer to write my widths/heights like this for readability:
```
min-height:
height:
max-height:
```
